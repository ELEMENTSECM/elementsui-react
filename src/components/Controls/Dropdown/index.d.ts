import * as React from "react";

export type DropdownDirection = "up" | "down" | "left" | "right";

export type DropdownAddonType = boolean | "prepend" | "append";

export interface DropdownProps {
	/**
	 * Is dropdown disabled
	 */
	disabled?: boolean;
	/**
	 * Open direction
	 */
	direction?: DropdownDirection;
	/**
	 * Is button group
	 */
	group?: boolean;
	/**
	 * Is dropdown open
	 */
	isOpen?: boolean;
	/**
	 * Is part of navigation menu
	 */
	nav?: boolean;
	/**
	 * Is dropdown active
	 */
	active?: boolean;
	/**
	 * Addon type
	 */
	addonType?: DropdownAddonType;
	/**
	 * Size
	 */
	size?: string;
	/**
	 * HTML element tag
	 */
	tag?: string;
	/**
	 * Toggle function
	 */
	toggle?: (...args: any[]) => any;
	/**
	 * Component children
	 */
	children?: React.ReactNode;
	/**
	 * Class name
	 */
	className?: string;
	/**
	 * Is in nav bar
	 */
	inNavbar?: boolean;
	/**
	 * Dropdoen is active if one of it's children is active
	 */
	setActiveFromChild?: boolean;
	/**
	 * Parent toggle
	 */
	parentToggle?: (...args: any[]) => any;
}

export default class Dropdown extends React.Component<DropdownProps, any> {
	render(): JSX.Element;
}
