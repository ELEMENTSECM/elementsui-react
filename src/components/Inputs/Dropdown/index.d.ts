import * as React from 'react';

export interface DropdownOptions {
	key: any;
	text: string;
	isSelected?: boolean;
}

export interface DropdownProps {
	/**
	 * DOM element id
	 */
	id?: string;
	/**
	 * Dropdown label
	 */
	label?: string;
	/**
	 * The key of the selected element
	 */
	selectedKey?: any;
	/**
	 * Default placeholder
	 */
	placeHolder?: string;
	/**
	 * Dropdown options
	 */
	options: DropdownOptions[];
	/**
	 * onChange event handler function
	 */
	onChange?: (...args: any[]) => any;
	/**
	 * User-defined styling
	 */
	getStyles?: (...args: any[]) => any;
}

export const Dropdown: React.SFC<DropdownProps>;
