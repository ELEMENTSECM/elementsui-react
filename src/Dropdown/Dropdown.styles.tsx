import styled from "styled-components";

export const DropdownContainer = styled.div`
	&.dropup,
	&.dropright,
	&.dropdown,
	&.dropleft {
		position: relative;
	}
	.dropdown-menu {
		position: absolute;
		top: 100%;
		left: 0;
		z-index: 110;
		display: none;
		float: left;
		min-width: 10rem;
		padding: 0.5rem 0;
		margin: 0.125rem 0 0;
		font-size: 1rem;
		color: inherit;
		text-align: left;
		list-style: none;
		background-color: inherit;
		background-clip: padding-box;
		border: 1px solid rgba(0, 0, 0, 0.15);
		border-radius: 0.25rem;
		&.show {
			display: block;
		}
	}
	.dropdown-menu-right {
		right: 0;
		left: auto;
	}
	@media (min-width: 576px) {
		.dropdown-menu-sm-right {
			right: 0;
			left: auto;
		}
	}
	@media (min-width: 768px) {
		.dropdown-menu-md-right {
			right: 0;
			left: auto;
		}
	}
	@media (min-width: 992px) {
		.dropdown-menu-lg-right {
			right: 0;
			left: auto;
		}
	}
	@media (min-width: 1200px) {
		.dropdown-menu-xl-right {
			right: 0;
			left: auto;
		}
	}
	.dropdown-menu-left {
		right: auto;
		left: 0;
	}
	@media (min-width: 576px) {
		.dropdown-menu-sm-left {
			right: auto;
			left: 0;
		}
	}
	@media (min-width: 768px) {
		.dropdown-menu-md-left {
			right: auto;
			left: 0;
		}
	}
	@media (min-width: 992px) {
		.dropdown-menu-lg-left {
			right: auto;
			left: 0;
		}
	}
	@media (min-width: 1200px) {
		.dropdown-menu-xl-left {
			right: auto;
			left: 0;
		}
	}
	&.dropup {
		.dropdown-menu {
			top: auto;
			bottom: 100%;
			margin-top: 0;
			margin-bottom: 0.125rem;
		}
		.dropdown-toggle::after {
			display: inline-block;
			margin-left: 0.255em;
			vertical-align: 0.255em;
			content: "";
			border-top: 0;
			border-right: 0.3em solid transparent;
			border-bottom: 0.3em solid;
			border-left: 0.3em solid transparent;
		}
		.dropdown-toggle:empty::after {
			margin-left: 0;
		}
	}
	&.dropright {
		.dropdown-menu {
			top: 0;
			right: auto;
			left: 100%;
			margin-top: 0;
			margin-left: 0.125rem;
		}
		.dropdown-toggle::after {
			display: inline-block;
			margin-left: 0.255em;
			vertical-align: 0.255em;
			content: "";
			border-top: 0.3em solid transparent;
			border-right: 0;
			border-bottom: 0.3em solid transparent;
			border-left: 0.3em solid;
		}
		.dropdown-toggle:empty::after {
			margin-left: 0;
		}
		.dropdown-toggle::after {
			vertical-align: 0;
		}
	}
	&.dropleft {
		.dropdown-menu {
			top: 0;
			right: 100%;
			left: auto;
			margin-top: 0;
			margin-right: 0.125rem;
		}
		.dropdown-toggle::after {
			display: inline-block;
			margin-left: 0.255em;
			vertical-align: 0.255em;
			content: "";
		}
		.dropdown-toggle::after {
			display: none;
		}
		.dropdown-toggle::before {
			display: inline-block;
			margin-right: 0.255em;
			vertical-align: 0.255em;
			content: "";
			border-top: 0.3em solid transparent;
			border-right: 0.3em solid;
			border-bottom: 0.3em solid transparent;
		}
		.dropdown-toggle:empty::after {
			margin-left: 0;
		}
		.dropdown-toggle::before {
			vertical-align: 0;
		}
	}

	.dropdown-divider {
		height: 0;
		margin: 0.5rem 0;
		overflow: hidden;
		border-top: 1px solid #e9ecef;
	}
	.dropdown-item {
		display: block;
		width: 100%;
		padding: 0.25rem 1.5rem;
		clear: both;
		font-weight: 400;
		color: #212529;
		text-align: inherit;
		white-space: nowrap;
		background-color: transparent;
		border: 0;
		&:first-child {
			border-top-left-radius: calc(0.25rem - 1px);
			border-top-right-radius: calc(0.25rem - 1px);
		}
		&:last-child {
			border-bottom-right-radius: calc(0.25rem - 1px);
			border-bottom-left-radius: calc(0.25rem - 1px);
		}
		&:hover,
		&:focus {
			color: #16181b;
			text-decoration: none;
			background-color: #f8f9fa;
		}
		&.active,
		&:active {
			color: #fff;
			text-decoration: none;
			background-color: #007bff;
		}
		&.disabled,
		&:disabled {
			color: #6c757d;
			pointer-events: none;
			background-color: transparent;
		}
	}
	.dropdown-item,
	.dropdown-header {
		font-size: 1.3rem;
	}
	.dropdown-header {
		display: block;
		padding: 0.5rem 1.5rem;
		margin-bottom: 0;
		font-family: OpenSans-Semibold, Tahoma;
		color: #fff;
		white-space: nowrap;
	}
	.dropdown-item-text {
		display: block;
		padding: 0.25rem 1.5rem;
		color: #212529;
	}
	.dropdown-toggle-split {
		padding-right: 0.5625rem;
		padding-left: 0.5625rem;
	}
	.dropdown-toggle-split::after,
	.dropup .dropdown-toggle-split::after,
	.dropright .dropdown-toggle-split::after {
		margin-left: 0;
	}
	.dropleft .dropdown-toggle-split::before {
		margin-right: 0;
	}
	.btn-sm + .dropdown-toggle-split,
	.btn-group-sm > .btn + .dropdown-toggle-split {
		padding-right: 0.375rem;
		padding-left: 0.375rem;
	}
	.btn-lg + .dropdown-toggle-split,
	.btn-group-lg > .btn + .dropdown-toggle-split {
		padding-right: 0.75rem;
		padding-left: 0.75rem;
	}
`;
