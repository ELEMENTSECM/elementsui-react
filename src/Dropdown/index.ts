export { default, DropdownProps } from "./Dropdown";
export { default as DropdownItem, DropdownItemProps } from "./DropdownItem";
export { default as DropdownMenu, DropdownMenuProps } from "./DropdownMenu";
export { default as DropdownToggle, DropdownToggleProps } from "./DropdownToggle";
