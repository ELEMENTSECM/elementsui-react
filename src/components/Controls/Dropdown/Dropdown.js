import * as React from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";
import { Manager } from "react-popper";
import classNames from "classnames";
import { omit, keyCodes } from "../../utils";

class Dropdown extends React.Component {
	getChildContext() {
		return {
			toggle: this.props.toggle,
			isOpen: this.props.isOpen,
			direction: this.props.direction === "down" && this.props.dropup ? "up" : this.props.direction,
			inNavbar: this.props.inNavbar
		};
	}

	componentDidMount() {
		this.handleProps();
	}

	componentDidUpdate(prevProps) {
		if (this.props.isOpen !== prevProps.isOpen) {
			this.handleProps();
		}
	}

	componentWillUnmount() {
		this.removeEvents();
	}

	getContainer = () => {
		return ReactDOM.findDOMNode(this);
	};

	addEvents = () => {
		["click", "touchstart", "keyup"].forEach(event =>
			document.addEventListener(event, this.handleDocumentClick, true)
		);
	};

	removeEvents = () => {
		["click", "touchstart", "keyup"].forEach(event =>
			document.removeEventListener(event, this.handleDocumentClick, true)
		);
	};

	handleDocumentClick = e => {
		if (e && (e.which === 3 || (e.type === "keyup" && e.which !== keyCodes.tab))) return;
		const container = this.getContainer();

		if (
			container.contains(e.target) &&
			container !== e.target &&
			(e.type !== "keyup" || e.which === keyCodes.tab)
		) {
			return;
		}

		this.toggle(e);
	};

	handleKeyDown = e => {
		if (
			keyCodes.tab === e.which ||
			(/button/i.test(e.target.tagName) && [keyCodes.space, keyCodes.enter].includes(e.which)) ||
			/input|textarea|select/i.test(e.target.tagName)
		) {
			return;
		}

		e.preventDefault();

		if (this.props.disabled) return;

		const container = this.getContainer();
		const isDropdown = e.target.classList.contains("dropdown-toggle");

		if (
			(e.which === keyCodes.space || e.which === keyCodes.enter) &&
			this.props.isOpen &&
			container !== e.target
		) {
			e.target.click();
		}

		if (
			e.which === keyCodes.esc ||
			(!this.props.isOpen && ![keyCodes.down, keyCodes.up].includes(e.which))
		) {
			this.toggle(e);
			container.querySelector("[aria-expanded]").focus();
			return;
		}

		const menuClass = "dropdown-menu";
		const itemClass = "dropdown-item";
		const disabledClass = "disabled";

		let items = container.querySelectorAll(`.${menuClass} .${itemClass}:not(.${disabledClass})`);
		items = Array.from(items).filter(v => v && v.offsetParent); // removes hidden elements..

		if (!items.length) return;

		let index = -1;

		const charPressed = String.fromCharCode(e.which).toLowerCase();

		for (let i = 0; i < items.length; i += 1) {
			const firstLetter = items[i].textContent && items[i].textContent[0].toLowerCase();
			if (firstLetter === charPressed || items[i] === e.target) {
				index = i;
				break;
			}
		}

		if (e.which === keyCodes.up && index > 0) {
			index -= 1;
		}

		if (e.which === keyCodes.down && index < items.length - 1) {
			index += 1;
		}

		if (index < 0) {
			index = 0;
		}

		items[index].focus();
	};

	handleProps = () => {
		if (this.props.isOpen) {
			this.addEvents();
		} else {
			this.removeEvents();
		}
	};

	toggle = e => {
		if (this.props.disabled) {
			return e && e.preventDefault();
		}

		return this.props.toggle(e);
	};

	render() {
		const {
			className,
			dropup,
			isOpen,
			group,
			size,
			nav,
			setActiveFromChild,
			active,
			addonType,
			...attrs
		} = omit(this.props, ["toggle", "disabled", "inNavbar", "direction"]);

		const direction = this.props.direction === "down" && dropup ? "up" : this.props.direction;

		attrs.tag = attrs.tag || (nav ? "li" : "div");

		let subItemIsActive = false;
		if (setActiveFromChild) {
			React.Children.map(this.props.children[1].props.children, dropdownItem => {
				if (dropdownItem && dropdownItem.props.active) subItemIsActive = true;
			});
		}

		const classes = classNames(
			className,
			"react-dropdown",
			direction !== "down" && `drop${direction}`,
			nav && active ? "active" : false,
			setActiveFromChild && subItemIsActive ? "active" : false,
			{
				[`input-group-${addonType}`]: addonType,
				"btn-group": group,
				[`btn-group-${size}`]: !!size,
				dropdown: !group && !addonType,
				show: isOpen,
				"nav-item": nav
			}
		);

		return (
			<Manager>
				<div {...attrs} className={classes} onKeyDown={this.handleKeyDown} role="menu" />
			</Manager>
		);
	}
}

Dropdown.propTypes = {
	/** Is dropdown disabled */
	disabled: PropTypes.bool,
	/** Open direction */
	direction: PropTypes.oneOf(["up", "down", "left", "right"]),
	/** Is button group */
	group: PropTypes.bool,
	/** Is dropdown open */
	isOpen: PropTypes.bool,
	/** Is part of navigation menu */
	nav: PropTypes.bool,
	/** Is dropdown active */
	active: PropTypes.bool,
	/** Addon type */
	addonType: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(["prepend", "append"])]),
	/** Size */
	size: PropTypes.string,
	/** HTML element tag */
	tag: PropTypes.string,
	/** Toggle function */
	toggle: PropTypes.func,
	/** Component children */
	children: PropTypes.node,
	/** Class name */
	className: PropTypes.string,
	/** Is in nav bar */
	inNavbar: PropTypes.bool,
	/** Dropdoen is active if one of it's children is active */
	setActiveFromChild: PropTypes.bool
};
Dropdown.defaultProps = {
	isOpen: false,
	direction: "down",
	nav: false,
	active: false,
	addonType: false,
	inNavbar: false,
	setActiveFromChild: false
};
Dropdown.childContextTypes = {
	toggle: PropTypes.func.isRequired,
	isOpen: PropTypes.bool.isRequired,
	direction: PropTypes.oneOf(["up", "down", "left", "right"]).isRequired,
	inNavbar: PropTypes.bool.isRequired
};

export default Dropdown;
