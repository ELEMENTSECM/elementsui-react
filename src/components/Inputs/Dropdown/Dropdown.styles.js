export const getStyles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Dropdown',
			{
				// place your styles here
			},
			className
		]
	};
};
