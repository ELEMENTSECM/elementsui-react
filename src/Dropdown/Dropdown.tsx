import * as React from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";
import { Manager } from "react-popper";
import classNames from "classnames";
import { keyCodes } from "../utils";
import { omit } from "lodash";
import DropdownToggle from "./DropdownToggle";
import DropdownMenu from "./DropdownMenu";
import DropdownItem from "./DropdownItem";
import { DropdownContainer } from "./Dropdown.styles";

export interface DropdownProps {
	/** Is dropdown disabled */
	disabled?: boolean;
	/** Open direction */
	direction?: "up" | "down" | "left" | "right";
	/** Is button group */
	group?: boolean;
	/** Is dropdown open */
	isOpen?: boolean;
	/** Is part of navigation menu */
	nav?: boolean;
	/** Is dropdown active */
	active?: boolean;
	/** Addon type */
	addonType?: "prepend" | "append" | boolean;
	/** Size */
	size?: string;
	/** HTML element tag */
	tag?: string;
	/** Toggle function */
	toggle: (e?: any) => void;
	/** Class name */
	className?: string;
	/** Is in nav bar */
	inNavbar?: boolean;
	/** Dropdoen is active if one of it's children is active */
	setActiveFromChild?: boolean;
	/**
	 * Parent toggle
	 */
	parentToggle?: (e?: any) => void;
}

export default class Dropdown extends React.Component<DropdownProps> {
	static Toggle = DropdownToggle;
	static Menu = DropdownMenu;
	static Item = DropdownItem;

	static defaultProps = {
		isOpen: false,
		direction: "down",
		nav: false,
		active: false,
		addonType: false,
		inNavbar: false,
		setActiveFromChild: false
	};

	static childContextTypes = {
		toggle: PropTypes.func.isRequired,
		isOpen: PropTypes.bool.isRequired,
		direction: PropTypes.oneOf(["up", "down", "left", "right"]).isRequired,
		inNavbar: PropTypes.bool.isRequired
	};

	getChildContext() {
		const toggle = e => {
			const isDropdownToggle = e.target.classList.contains("dropdown-toggle");
			if (this.props.parentToggle && this.props.isOpen && !isDropdownToggle) {
				this.props.parentToggle(e);
			}
			this.props.toggle(e);
		};
		return {
			toggle: toggle,
			isOpen: this.props.isOpen,
			direction: this.props.direction,
			inNavbar: this.props.inNavbar
		};
	}

	componentDidMount() {
		this.handleProps();
	}

	componentDidUpdate(prevProps) {
		if (this.props.isOpen !== prevProps.isOpen) {
			this.handleProps();
		}
	}

	componentWillUnmount() {
		this.removeEvents();
	}

	getContainer = () => {
		return ReactDOM.findDOMNode(this) as Element;
	};

	addEvents = () => {
		["click", "touchstart", "keyup"].forEach(event =>
			document.addEventListener(event, this.handleDocumentClick, true)
		);
	};

	removeEvents = () => {
		["click", "touchstart", "keyup"].forEach(event =>
			document.removeEventListener(event, this.handleDocumentClick, true)
		);
	};

	handleDocumentClick = e => {
		if (e && (e.which === 3 || (e.type === "keyup" && e.which !== keyCodes.tab))) return;
		const container = this.getContainer();

		if (
			container &&
			container.contains(e.target) &&
			container !== e.target &&
			(e.type !== "keyup" || e.which === keyCodes.tab)
		) {
			return;
		}

		this.toggle(e);
	};

	handleKeyDown = e => {
		if (
			keyCodes.tab === e.which ||
			(/button/i.test(e.target.tagName) && [keyCodes.space, keyCodes.enter].includes(e.which)) ||
			/input|textarea|select/i.test(e.target.tagName)
		) {
			return;
		}

		e.preventDefault();

		if (this.props.disabled) return;

		const container = this.getContainer();

		if (
			(e.which === keyCodes.space || e.which === keyCodes.enter) &&
			this.props.isOpen &&
			container !== e.target
		) {
			e.target.click();
		}

		if (
			e.which === keyCodes.esc ||
			(!this.props.isOpen && ![keyCodes.down, keyCodes.up].includes(e.which))
		) {
			this.toggle(e);
			if (container) {
				const expanded = container.querySelector("[aria-expanded]") as HTMLElement;
				expanded && expanded.focus();
			}
			return;
		}

		const menuClass = "dropdown-menu";
		const itemClass = "dropdown-item";
		const disabledClass = "disabled";

		const items = Array.from(
			container.querySelectorAll(`.${menuClass} .${itemClass}:not(.${disabledClass})`)
		).filter((v: any) => v && v.offsetParent) as HTMLElement[];

		if (!items.length) return;

		let index = -1;

		const charPressed = String.fromCharCode(e.which).toLowerCase();

		for (let i = 0; i < items.length; i += 1) {
			const firstLetter = items[i].textContent && items[i].textContent![0].toLowerCase();
			if (firstLetter === charPressed || items[i] === e.target) {
				index = i;
				break;
			}
		}

		if (e.which === keyCodes.up && index > 0) {
			index -= 1;
		}

		if (e.which === keyCodes.down && index < items.length - 1) {
			index += 1;
		}

		if (index < 0) {
			index = 0;
		}

		items[index].focus();
	};

	handleProps = () => {
		if (this.props.isOpen) {
			this.addEvents();
		} else {
			this.removeEvents();
		}
	};

	toggle = e => {
		if (this.props.disabled) {
			return e && e.preventDefault();
		}

		return this.props.toggle(e);
	};

	render() {
		const {
			className,
			isOpen,
			group,
			size,
			nav,
			setActiveFromChild,
			active,
			addonType,
			direction,
			...attrs
		} = omit(this.props, ["toggle", "parentToggle", "disabled", "inNavbar"]);

		attrs.tag = attrs.tag || (nav ? "li" : "div");

		let subItemIsActive = false;
		if (setActiveFromChild) {
			React.Children.map((this.props.children as any[])[1].props.children, dropdownItem => {
				if (dropdownItem && dropdownItem.props.active) subItemIsActive = true;
			});
		}

		const classes = classNames(
			className,
			"react-dropdown",
			direction !== "down" && `drop${direction}`,
			nav && active ? "active" : false,
			setActiveFromChild && subItemIsActive ? "active" : false,
			{
				[`input-group-${addonType}`]: addonType,
				"btn-group": group,
				[`btn-group-${size}`]: !!size,
				dropdown: !group && !addonType,
				show: isOpen,
				"nav-item": nav
			}
		);

		return (
			<Manager>
				<DropdownContainer
					{...attrs}
					className={classes}
					onKeyDown={this.handleKeyDown}
					role={this.props.inNavbar ? "presentation" : "menu"}
				/>
			</Manager>
		);
	}
}
