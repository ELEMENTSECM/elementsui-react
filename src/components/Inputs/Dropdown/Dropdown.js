import * as React from 'react';
import PropTypes from 'prop-types';
import { Dropdown as UIFabDropdown } from 'office-ui-fabric-react/lib/Dropdown';

/** Dropdown example */
function Dropdown({ id, label, selectedKey, placeHolder, options, onChange }) {
	return (
		<UIFabDropdown
			id={id}
			label={label}
			selectedKey={selectedKey}
			placeHolder={placeHolder}
			options={options}
			onChanged={onChange}
		/>
	);
}

Dropdown.propTypes = {
	/** DOM element id */
	id: PropTypes.string,
	/** Dropdown label */
	label: PropTypes.string,
	/** The key of the selected element */
	selectedKey: PropTypes.any,
	/** Default placeholder */
	placeHolder: PropTypes.string,
	/** Dropdown options */
	options: PropTypes.arrayOf(
		PropTypes.shape({
			/** Option key value */
			key: PropTypes.any.isRequired,
			/** Option text value */
			text: PropTypes.string.isRequired,
			/** Option is selected */
			isSelected: PropTypes.bool
		})
	).isRequired,
	/** onChange event handler function */
	onChange: PropTypes.func
};

export default Dropdown;
