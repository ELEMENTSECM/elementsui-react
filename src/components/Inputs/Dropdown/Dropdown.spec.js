import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';

describe('Dropdown', () => {
	const options = [
		{
			key: 1,
			text: 'Option 1'
		},
		{
			key: 2,
			text: 'Option 2'
		},
		{
			key: 3,
			text: 'Option 3',
			isSelected: true
		}
	];

	test('displays correctly', () => {
		const tree = renderer
			.create(
				<Dropdown
					htmlId="testDropdown"
					label="test dropdown"
					selectedKey="3"
					placeHolder="Select test..."
					options={options}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
