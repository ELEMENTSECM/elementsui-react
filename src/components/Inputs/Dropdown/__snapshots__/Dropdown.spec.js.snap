// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown displays correctly 1`] = `
<div
  className="ms-Dropdown-container"
>
  <label
    className="ms-Label ms-Dropdown-label root-36"
    htmlFor="testDropdown"
    id="testDropdown-label"
    required={undefined}
  >
    test dropdown
  </label>
  <div
    aria-activedescendant={null}
    aria-autocomplete="none"
    aria-describedby="testDropdown-option"
    aria-disabled={undefined}
    aria-expanded="false"
    aria-label={undefined}
    aria-live="assertive"
    aria-owns={null}
    className="ms-Dropdown root_46f1b9db"
    data-is-focusable={true}
    id="testDropdown"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="combobox"
    tabIndex={0}
  >
    <span
      aria-atomic={true}
      aria-readonly="true"
      className="ms-Dropdown-title title_46f1b9db ms-Dropdown-titleIsPlaceHolder titleIsPlaceHolder_46f1b9db"
      id="testDropdown-option"
      role="listbox"
    >
      <span>
        Select test...
      </span>
    </span>
    <span
      className="ms-Dropdown-caretDownWrapper caretDownWrapper_46f1b9db"
    >
      <i
        aria-hidden={true}
        className="root-37 ms-Dropdown-caretDown caretDown_46f1b9db"
        data-icon-name="ChevronDown"
        role="presentation"
      />
    </span>
  </div>
</div>
`;
